google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/benchmark/Benchmark.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery.benchmark;
+
+import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryResponse;
+import java.io.FileInputStream;
+import java.util.List;
+import org.threeten.bp.Clock;
+import org.threeten.bp.Duration;
+import org.threeten.bp.Instant;
+
+public class Benchmark {
+  public static void main(String[] args) throws Exception {
+    if (args.length < 1) {
+      System.out.println("need path to queries.json");
+      return;
+    }
+    String[] requests =
+        new JacksonFactory()
+            .createJsonParser(new FileInputStream(args[0]))
+            .parseAndClose(String[].class);
+
+    Clock clock = Clock.systemUTC();
+    BigQuery bq = BigQueryOptions.getDefaultInstance().getService();
+
+    for (String request : requests) {
+      if (request.isEmpty()) {
+        continue;
+      }
+
+      Instant start = clock.instant();
+      QueryResponse queryResponse =
+          bq.query(QueryJobConfiguration.newBuilder(request).setUseLegacySql(false).build());
+
+      int rows = 0;
+      int cols = 0;
+      Duration firstByte = null;
+      for (List<FieldValue> row : queryResponse.getResult().iterateAll()) {
+        rows++;
+        if (cols == 0) {
+          cols = row.size();
+          firstByte = Duration.between(start, clock.instant());
+        } else if (cols != row.size()) {
+          throw new IllegalStateException(
+              String.format("expected %d cols, found %d", cols, row.size()));
+        }
+      }
+      Duration total = Duration.between(start, clock.instant());
+
+      double firstByteSec = (double)(firstByte.getNano())/1000/1000/1000 + firstByte.getSeconds();
+      double totalSec = (double)(total.getNano())/1000/1000/1000 + total.getSeconds();
```

Comment:
Could we use `Duration` for more readable time calculations?

Reply:
We're already using `Duration` right? The lines don't "create new durations", it just turns the existing durations into second form so they can be reported.

Reply:
So why not use `firstByte.getSeconds()` and `total.getSeconds()`?

Reply:
We do call `getSeconds` at the end of the statement. It only reports whole seconds though. If your duration is 12.5s, `getSeconds` will only report the 12. You need to get the 0.5 from `getNano` which is what we're doing here.

Reply:
Oh derp, I didn't look close enough. 

I don't like the error-prone repetition of 1000. Maybe at least make a constant out of the divisor.

Reply:
Done